sound/soc/sof/ipc4-loader.c

```diff
@@ -423,6 +423,53 @@ int sof_ipc4_reload_fw_libraries(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+/**
+ * sof_ipc4_update_cpc_from_manifest - Update the cpc in base config from manifest
+ * @sdev: SOF device
+ * @fw_module: pointer struct sof_ipc4_fw_module to parse
+ * @basecfg: Pointer to the base_config to update
+ */
+void sof_ipc4_update_cpc_from_manifest(struct snd_sof_dev *sdev,
+				       struct sof_ipc4_fw_module *fw_module,
+				       struct sof_ipc4_base_module_cfg *basecfg)
+{
+	const struct sof_man4_module_config *fw_mod_cfg;
+	u32 cpc_pick = 0;
+	u32 max_cpc = 0;
+	int i;
+
+	/* The CPC is set from topology, ignoring manifest */
+	if (basecfg->cpc)
+		return;
+
+	if (!fw_module || !fw_module->fw_mod_cfg)
+		return;
+
+	fw_mod_cfg = fw_module->fw_mod_cfg;
+	for (i = 0; i < fw_module->man4_module_entry.cfg_count; i++) {
+		if (basecfg->obs == fw_mod_cfg[i].obs &&
+		    basecfg->ibs == fw_mod_cfg[i].ibs &&
+		    cpc_pick < fw_mod_cfg[i].cpc)
+			cpc_pick = fw_mod_cfg[i].cpc;
```

Comment:
cpc_pick is initialized to zero and is an u32, it will by construction always be smaller than any other value.

This initialization will never be performed.

Reply:
I'm not following you.
fw_mod_cfg[i].cpc is also u32

cpc_pick = 0;
fw_mod_cfg[0].cpc = 0;
fw_mod_cfg[1].cpc = 10;
fw_mod_cfg[2].cpc = 5;
fw_mod_cfg[3].cpc = 50;

cpc_pick will be 50 after the loop:
i = 0;
  cpc_pick == fw_mod_cfg[i].cpc -> no change in cpc_pick
i = 1;
  cpc_pick < fw_mod_cfg[i].cpc -> cpc_pick = 10;
i = 2;
  cpc_pick > fw_mod_cfg[i].cpc -> no change in cpc_pick
i = 3;
  cpc_pick < fw_mod_cfg[i].cpc ->cpc_pick = 50;

Reply:
agree with @ujfalusi , @plbossart could you double-check?

Reply:
never mind, I read this sideways. stupid me.